src/test/java/guitests/BoardAutoCreatorTest.java

```diff
@@ -105,14 +105,22 @@ public void sampleBoardAutoCreationTest() {
         waitUntilNodeAppears(SAMPLE_BOARD_DIALOG);
         click("OK");
 
-        assertEquals(panelControl.getPanelCount(), SAMPLE_PANEL_NAMES.size());
+        assertEquals(panelControl.getPanelCount(), BoardAutoCreator.getSamplePanelDetails().size());
         assertEquals(panelControl.getCurrentlySelectedPanel(), Optional.of(0));
         assertEquals(panelControl.getNumberOfSavedBoards(), 1);
+        verifySampleBoard(panelControl, BoardAutoCreator.getSamplePanelDetails());
+    }
 
-        List<PanelInfo> panelInfos = panelControl.getCurrentPanelInfos();
-        for (int i = 0; i < SAMPLE_PANEL_NAMES.size(); i++){
-            assertEquals(panelInfos.get(i).getPanelFilter(), SAMPLE_PANEL_FILTERS.get(i));
-            assertEquals(panelInfos.get(i).getPanelName(), SAMPLE_PANEL_NAMES.get(i));
+    /**
+     * Tests whether the sample panels along with their filters are rightly created and present in the UI
+     */
+    public static void verifySampleBoard(PanelControl pc, Map<String, String> samplePanelDetails){
```

Comment:
1. Why does the method take `PanelControl` as the parameter instead of `pc.getCurrentPanelInfos()`?
2. If you change it to take `pc.getCurrentPanelInfos()`, the method becomes more general, so the name/comment should not mention 'sample board'

Alternatively, you could feed both `PanelControl` and `BoardAutoCreator` into this method in which case all asserts should be done inside method. At the moment, some asserts are done outside and some inside.